src/Tasks/Microsoft.NET.Build.Tasks/DependencyContextBuilder2.cs

```diff
@@ -0,0 +1,858 @@
+                serviceable: serviceable);
+
+            return runtimeLibrary;
+        }
+
+        private CompilationLibrary GetCompilationLibrary(DependencyLibrary library)
+        {
+            GetCommonLibraryProperties(library,
+                out string hash,
+                out HashSet<Dependency> libraryDependencies,
+                out bool serviceable,
+                out string path,
+                out string hashPath,
+                out SingleProjectInfo referenceProjectInfo);
+
+            List<string> assemblies = new List<string>();
+
+            if (library.Type == "project" && !(referenceProjectInfo is UnreferencedProjectInfo))
+            {
+                assemblies.Add(referenceProjectInfo.OutputName);
+            }
+            else if (_compileReferences != null && _compileReferences.TryGetValue(library.Name, out var compileReferences))
+            {
+                foreach (var compileReference in compileReferences)
+                {
+                    assemblies.Add(compileReference.PathInPackage);
+                }
+            }
+
+            return new CompilationLibrary(
+                type: library.Type,
+                name: library.Name,
+                version: library.Version.ToString(),
+                hash,
+                assemblies,
+                libraryDependencies,
+                serviceable,
+                path,
+                hashPath);
+        }
+
+        private void GetCommonLibraryProperties(DependencyLibrary library,
+                    out string hash,
+                    out HashSet<Dependency> dependencies,
+                    out bool serviceable,
+                    out string path,
+                    out string hashPath,
+                    out SingleProjectInfo referenceProjectInfo)
+        {
+            serviceable = true;
+            referenceProjectInfo = null;
+
+            dependencies = new HashSet<Dependency>();
+            List<LibraryDependency> libraryDependencies;
+            if (_libraryDependencies.TryGetValue(library.Name, out libraryDependencies))
+            {
+                foreach (var dependency in libraryDependencies)
+                {
+                    if (_dependencyLibraries.TryGetValue(dependency.Name, out var libraryDependency))
+                    {
+                        if (!libraryDependency.ExcludeFromRuntime ||
+                            (!libraryDependency.ExcludeFromCompilation && IncludeCompilationLibraries))
+                        {
+                            dependencies.Add(libraryDependency.Dependency);
+                        }
+                    }
+                }
+            }
+
+            hash = string.Empty;
+            path = null;
+            hashPath = null;
+            if (library.Type == "package")
+            {
+                // TEMPORARY: All packages are serviceable in RC2
```

Comment:
Nit: is this temporary comment from 2016 still relevant today?

Reply:
I don't think so, but I wasn't sure so I figured it was best to preserve it in the re-implementation.

Reply:
@eerhardt

Reply:
Do we have any test coverage for this servicing mechanism?

Reply:
This currently doesn't work because https://github.com/NuGet/Home/issues/2923 hasn't been implemented yet.